### PR TITLE
Nano branches from ValueMap of elaborated types

### PR DESCRIPTION
**Pull request by vlimant**
#### PR description:

in order to enable table extension, using elaborated types (other than int, float, ...) some specific templated producer need to be added. no change of existing branch/module are introduced. 

#### PR validation:

the nano matrix succeeded.

### CODE REVIEW

File: PhysicsTools/NanoAOD/interface/SimpleFlatTableProducer.h
Comment (vlimant) [inline PhysicsTools/NanoAOD/interface/SimpleFlatTableProducer.h:125]: it failes on passing `cc` around ; any clue anyone ?

**Comment by vlimant** [comment]
meant to help with #43525

**Comment by vlimant** [comment]
enable nano

**Comment by vlimant** [comment]
please test

**Comment by vlimant** [comment]
please test

**Review by mbluj — CHANGES_REQUESTED** [review]
@vlimant Thank you for this development. I tested it with my setup and it works fine after implementing a few fixes here and there (see directly in code & one below).
How I should proceed? Should I introduce those changes into my PR or this one is added and then my development branch is rebased on top of it? The former is simpler for me.

L345 & L352:
`if (!extvars_.empty())` => `if (!extvars_.empty() || !typedextvars_.empty())`

**Comment by mbluj** [comment]
@vlimant Hi my comments are partly outdated as were given to previous version of the development, but some are still valid.

File: PhysicsTools/NanoAOD/interface/SimpleFlatTableProducer.h
Comment (vlimant) [reply PhysicsTools/NanoAOD/interface/SimpleFlatTableProducer.h:441]: thanks indeed

File: PhysicsTools/NanoAOD/interface/SimpleFlatTableProducer.h
Comment (vlimant) [reply PhysicsTools/NanoAOD/interface/SimpleFlatTableProducer.h]: the intermediate template type is actually not necessary

**Comment by vlimant** [comment]
please test

**Comment by vlimant** [comment]
@mbluj : I think that the best way is to merge this, and you rebase #43525 to pick up and use the feature (removing `SimpleCandidate2CandidateFlatTableProducer`) along the way

**Comment by mbluj** [comment]
> @mbluj : I think that the best way is to merge this, and you rebase #43525 to pick up and use the feature (removing `SimpleCandidate2CandidateFlatTableProducer`) along the way

OK, so I am waiting until this is merged. Could it be also backported to 14_0?

**Comment by vlimant** [comment]
+1

**Comment by mbluj** [comment]
Many thanks @vlimant! I am moving my development on top of it.
Could this development be also backported to 14_0 series, please?

**Comment by antoniovilela** [comment]
+1

**Comment by mbluj** [comment]
For reference: backported to 14_0_X in #44040.

**Pull request merged**
